### ISSUE TITLE
java.lang.NullPointerException if using '@Test(expected = <>)' argument

### ISSUE THREAD

**Issue by bebef1987** [serenity-bdd/serenity-core#858]
if we are using the `expected` argument of '@Test' tag then we get a java.lang.NullPointerException error:

```
java.lang.NullPointerException
	at net.thucydides.core.model.TestOutcome$StepResetBuilder.causedBy(TestOutcome.java:2316)
	at net.thucydides.core.model.TestOutcome.resetFailingStepsCausedBy(TestOutcome.java:721)
	at net.thucydides.core.steps.BaseStepListener.exceptionExpected(BaseStepListener.java:131)
	at net.thucydides.core.steps.StepEventBus.exceptionExpected(StepEventBus.java:648)
	at net.thucydides.junit.listeners.JUnitStepListener.updateResultsForExpectedException(JUnitStepListener.java:111)
	at net.thucydides.junit.listeners.JUnitStepListener.updateResultsUsingTestAnnotations(JUnitStepListener.java:106)
	at net.thucydides.junit.listeners.JUnitStepListener.testFinished(JUnitStepListener.java:96)
	at org.junit.runner.notification.SynchronizedRunListener.testFinished(SynchronizedRunListener.java:56)
	at org.junit.runner.notification.RunNotifier$7.notifyListener(RunNotifier.java:190)
	at org.junit.runner.notification.RunNotifier$SafeNotifier.run(RunNotifier.java:72)
	at org.junit.runner.notification.RunNotifier.fireTestFinished(RunNotifier.java:187)
	at org.junit.internal.runners.model.EachTestNotifier.fireTestFinished(EachTestNotifier.java:38)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:331)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at net.serenitybdd.junit.runners.SerenityRunner.performRunChild(SerenityRunner.java:471)
	at net.serenitybdd.junit.runners.SerenityRunner.runChild(SerenityRunner.java:446)
	at net.serenitybdd.junit.runners.SerenityRunner.runChild(SerenityRunner.java:55)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at net.serenitybdd.junit.runners.SerenityRunner.run(SerenityRunner.java:252)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
```

**Comment by bebef1987** [comment]
https://github.com/bebef1987/demo-project here you can find a demo project

**Comment by wakaleo** [comment]
Fixed in 1.5.4-rc.2
